libs/langgraph/src/pregel/runner.ts

```diff
@@ -333,10 +409,15 @@ export class PregelRunner {
     let startedTasksCount = 0;
 
     let listener: () => void;
-    const signalPromise = new Promise<never>((_resolve, reject) => {
+
+    const abortPromise = new Promise<never>((_resolve, reject) => {
       listener = () => reject(new Error("Abort"));
-      signal?.addEventListener("abort", listener);
-    }).finally(() => signal?.removeEventListener("abort", listener));
+      signals?.timeoutAbortSignal?.addEventListener("abort", listener);
+      signals?.externalAbortSignal?.addEventListener("abort", listener);
+    }).finally(() => {
+      signals?.timeoutAbortSignal?.removeEventListener("abort", listener);
+      signals?.externalAbortSignal?.removeEventListener("abort", listener);
+    });
```

Comment:
could've just done `{ once: true }` as the 3rd arg here, but I wanted to avoid a double call to `reject`.

Reply:
Ah, although I could compose them and then ditch the `finally` block - I'll do that instead.